### ISSUE TITLE
slower navbarPage build in 0.11?

### ISSUE THREAD

**Issue by mostly-harmless** [rstudio/shiny#693]
It seems page build is quite a bit slower in 0.11 compared to 0.10.2.2. I wondered if it might be related to issue #373 I had when I started using navbarPage but some initial profiling (see screenshot) suggests it is probably something else. The page/panel where the difference is most obvious is Regression > Linear (OLS). A server version of my app running 0.10.2.2 is here: http://vnijs.rady.ucsd.edu:3838/marketing/

Code is here (see the quant or marketing apps in inst/): https://github.com/mostly-harmless/radiant

Note that I put a `stop` at the top of inst/base/global for now `if(packageVersion('shiny') > "0.10.2.2") stop("...")` until I have time to make the css compatible with bootstrap 3.

![image](https://cloud.githubusercontent.com/assets/2998956/5794052/100e87d6-9f12-11e4-9b98-13178f0ea88e.png)

**Comment by wch** [comment]
I'm trying to replicate the issue. When I comment out the version check and try to run the marketing app, it gives an error:

```
Error in eval(expr, envir, enclos) : object 'opts_chunk' not found 
```


I searched for `opts_chunk` and added `knitr::` to the front of it. Then I got these warnings:

```
Warning: Unhandled error in observer: could not find function "%>%"
```

Running `library(magrittr)` from the console made those go away.  After that, the app runs, but I get this in the browser:

```
Error: could not find function "slice"
Error: could not find function "markdownToHTML"
```

Going to Regression -> Linear (OLS) does work, though. I have to say that it doesn't seem significantly different from the live version you linked to.

Can you provide more details on how to replicate the issue?

**Comment by mostly-harmless** [comment]
I'm putting radiant into an R-package structure and it seems not all dependencies are being loaded at the moment. You can install all packages you would need from a non-CRAN repo (including Shiny .10.2.2). To install from that repo use:

options(repos = c(XRAN = "http://mostly-harmless.github.io/radiant_miniCRAN/"))

and use:

devtools::source_url("https://raw.githubusercontent.com/mostly-harmless/radiant_miniCRAN/gh-pages/pkgs.R")
sapply(pkgs, require, lib.loc = local_dir, character.only=TRUE)

to load everything

I don't know the best way to profile/isolate the build speed of the regression page but in Chrome or Safari on a mac there really is a substantial difference. From just looking at the page profile for Shiny .10.2.2 (see image below) perhaps there is something going on with extend.camelCase?

![image](https://cloud.githubusercontent.com/assets/2998956/5846556/5961e13a-a179-11e4-8dc0-57440112a0bb.png)

**Comment by mostly-harmless** [comment]
I pushed some updates to GitHub that should make installing/running Radiant easier. Paste the commands below in the (Rstudio) console:

options(repos = c(XRAN = 'http://mostly-harmless.github.io/radiant_miniCRAN/'))
install.packages('radiant')

When the install is done try:

library(radiant)
radiant()

That should pull in all dependencies. I also changed the 'stop' for 0.11 to a 'warning' so comparing to 0.10.2.2 should be a bit easier I hope.

**Comment by wch** [comment]
Thanks for the simplified setup - it's very helpful for debugging this issue.

One big hint: When the page is loaded, the `sendImageSize` function gets called about 2000 times. When I go to Regression->OLS, it gets called about 4500 more times! After that, switching between the original Data page and Regression->OLS results in about 500 calls to `sendImageSize` each time.

**Comment by mostly-harmless** [comment]
Interesting. This is for .11 I assume and not with .10.22? sendImageSize is a command Shiny uses correct? It is not in my code anywhere. This is not related to mathJax by any chance is it (https://github.com/rstudio/shiny/issues/692)?





—

**Comment by mostly-harmless** [comment]
I just noticed #717. The issue reported there is consistent with my guess that the regression > ols page was loading slowly because it has more input elements (and conditions) on it than other pages.

**Comment by wch** [comment]
Yes, that's with 0.11. I haven't tested 0.10.2.2, but I think that the old version calls `sendImageSize` 1/7 as many times. Even that is too many times. I hope that once we have a fix in, it'll be reduced to a handful of times.

**Comment by wch** [comment]
We merged a fix for this into master (#723). If you could reinstall and test it out, we'd appreciate it! Please let us know about performance and if anything odd happens.

```R
devtools::install_github('rstudio/shiny')
```

**Comment by mostly-harmless** [comment]
Works great. Thanks for the fix!!


—
